### PR TITLE
Allow Powershell provisioner to use service accounts

### LINKED ISSUE #5478: Run Puppet Masterless provisioner as elevated user on Windows
When using the Puppet Masterless on Windows it fails when trying to install SQL Server as it needs administrative privileges. I always thought that cmd commands would execute in an elevated command prompt? Is this not the case? At least on Windows, installations for a good amount of tools do require administrative rights and this would prohibit that.

The work around for now is to execute Puppet from a elevated powershell script.

- Packer version: 1.1.0
- Windows 7 / Ubuntu 16.04

### LINKED ISSUE #6104: Allow users of the PowerShell provisioner to use service accounts for elevated_user
Certain applications may require an interactive session on Windows in order to function properly. WinRM commands are not run in an interactive session, and currently the only way to accomplish this is to run the application from a scheduled task as a service account such as the `SYSTEM` user.

One way to allow the user to do this could be to allow the `elevated_password` to be empty, and use the service account logon type if an empty string is passed. In order to schedule a task as the `SYSTEM` user, `RegisterTaskDefinition` should be passed a [logonType](https://msdn.microsoft.com/en-us/library/windows/desktop/aa382577) of [`5` (`TASK_LOGON_SERVICE_ACCOUNT`)](https://msdn.microsoft.com/en-us/library/windows/desktop/aa382075) and `password` of `$null`. Additionally, the `<LogonType>` tag should be omitted from the task definition xml. See https://github.com/WinRb/winrm-elevated/blob/master/lib/winrm-elevated/scripts/elevated_shell.ps1#L10.

### PR DESCRIPTION
**Pull request by bodgit**
This uses the logic defined in #6104 to allow running Powershell with a service account, i.e. setting `elevated_password` to an empty value.

Tested using a Powershell script that runs Puppet with `elevated_user` set to `SYSTEM`. Previously Puppet didn't have sufficient permissions within the WinRM connection to install all packages/apply changes whereas now it seems to work properly. Ultimately I'd like to extract some of this elevated functionality to fix #5478 (except with the `puppet-server` provisioner instead) however my familiarity with Windows internals is a tad lacking currently.

Fixes #6104

### CODE REVIEW

**Comment by azr** [comment]
Nice one ! Thanks.
The test for what you changed are failing now.

**Comment by bodgit** [comment]
Ok, I'll try and get that fixed up.

**Comment by bodgit** [comment]
Tests seem happy now.

File: provisioner/powershell/provisioner.go [hunk 1]
```diff
@@ -190,11 +190,6 @@ func (p *Provisioner) Prepare(raws ...interface{}) error {
 			errors.New("Only one of script or scripts can be specified."))
 	}
 
-	if p.config.ElevatedUser != "" && p.config.ElevatedPassword == "" {
```
[the remark this answers is not in the document]
Comment (bodgit) [reply provisioner/powershell/provisioner.go]: There's still the check below it that if the elevated password is specified then the user should also be, but if the password is now optional then this check isn't valid. AFAIK these config options don't have a `nil` value so I don't see how checking `p.config.ElevatedUser != ""` is useful now.

[the remark this answers is not in the document]
Comment (azr) [reply provisioner/powershell/provisioner.go]: Oh right, I see it down below 🙂

**Pull request merged**